### ISSUE TITLE
Suggestion: Add features to the FunctionRouter

### ISSUE THREAD

**Issue by ferblaca** [spring-cloud/spring-cloud-stream#2281]
We have been analyzing this #1790, and we see a clear backward step in functionality with `FunctionRouter` versus the filter `conditions` in `@StreamListener`. 

These are the limitations we found with the `FunctionRouter`:

- Only one `FunctionRouter` can be defined per application.
- The `FunctionRouter` can only belong to one Binder: for example, and although it is rare, if your application consumes Rabbit and Kafka, the `FunctionRouter`has to be configured with one of the binders, it will never be able to read two different data sources.
- The `FunctionRouter` does not know the origin of the message consumed and therefore, even if it consumes from several topics, you can not customize a `routing-expression` depending on the origin of the message.

For all this, we see that the `FunctionRouter` is another functionality totally different from the filtering in the `@StreamListener`, which is a significant loss in terms of usability in Spring Cloud Stream with the functional approach.

We believe it would be important to reconsider this functionality and add at least some of the following features to the `FunctionRouter`:

- Being able to define more than one `FunctionRouter` per application.
- To be able to define one `routing-expression` per message origin. For example, if the `FunctionRouter` has configured **topic1** and **topic2** destinations, it would be nice to be able to define one `routing-expression` for **topic1** messages and another `routing-expression` for **topic2** messages.

Thank you very much, we hope that the request will be considered.

**Comment by ferblaca** [comment]
Hi @olegz ! thank you very much for your quick reply.

Yes, as you say it is possible to define more than one `FunctionRouter`, and therefore assign more than one binder per application. The only thing is that all the defined `FunctionRouter` share by default the same bean `FunctionProperties` and therefore the same routing expression. Would it be possible to assign by configuration (.yml or properties) a routing expression to each defined `FunctionRouter`?

Regarding the origin of the message: Yes, it is a request, if the topic/queue from which the message has been consumed were known it could be very useful to compose one expression or another depending on this value, always talking about the case in which the `FunctionRouter` has several destinations configured.

**Comment by olegz** [comment]
Makes sense, i'll change the title of an issue and mark it as feature request
